### PR TITLE
add functionality to check if an object is included in another object

### PR DESCRIPTION
**Pull request by rafmagana**
Some days ago I was using should.js to test an API, I needed to test against a JSON object which was a response from the API, there was some properties/values in the response I didn't care about, look at this case:

Reponse

```
  var response = { 'attr1':'value1': 'attr2':'value2'}
```

Wanted to test if the properties/values of this object were part of the response:

```
var my_local_obj = { 'attr1':'value1'}
```

should.include and includeEql to the rescue, right? well, no

```
  response.should.include(my_local_object); // it only works with strings, elements of an array, etc
  response.should.includeEql(my_local_object); // It only works with arrays
```

so I made this patch to support just that, hope you find it interesting.

maybe I should have added this feature to the includeEql() method, I just think includeEql() is confusing, I'd like to have the include() and includeEql() functionality in the same method but I'd like to know what you think.

Thanks.

### CODE REVIEW

**Comment by rafmagana** [comment]
any comment?

File: lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+var Helpers = {};
+
+// Stolen from https://gist.github.com/1251001/a228117756e2e9eed2d31ed1077a5a4353d997fb
+Helpers.areEqual = function (a, b) {
```
Comment (tj) [inline lib/helpers.js]: the rest looks fine but why do we need this? what about node's regular assert.deepEqual() stuff

Comment (rafmagana) [inline lib/helpers.js]: well, the thing is that `assert.deepEqual` throws an error when the assertion fails:

```
> assert.deepEqual({lorem:1,ipsum:2}, {lorem:1});

AssertionError: {"lorem":1,"ipsum":2} deepEqual {"lorem":1}
    at [object Context]:1:8
```

and I just wanted a function to return true or false, I could use try/catch though, what do you think is better?

Comment (tj) [inline lib/helpers.js]: try/catch would be fine, or if we can access that function from the assert module that would be better but im not sure if they expose it or not

Comment (tj) [inline lib/helpers.js]: right right we already have this in ./lib/eql.js since they didnt expose it

Comment (rafmagana) [inline lib/helpers.js]: nice!! let me do the switch, thanks.

**Comment by rafmagana** [comment]
what do you think now?

**Comment by tj** [comment]
looks a lot better, ill comment inline

File: Readme.md [hunk 2]
```diff
@@ -255,6 +255,14 @@ Assert substring:
     'foo bar baz'.should.include('baz')
     'foo bar baz'.should.not.include('FOO')
 
+Assert object includes another object:
+
+    var full_object = {'foo':'bar', 'baz': 'qux'}
+    var partial_object = {'foo':'bar'}
+
+    full_object.should.include(partial_object)
```
Comment (tj) [inline Readme.md]: for tests and examples we should use camel-case, also we dont need to be so verbose,
ideally something descriptive like:

``` js
var tobi = { name: 'Tobi' };
var user = { pet: tobi, name: 'TJ' };
user.should.include({ pet: tobi });
```

File: lib/should.js [hunk 3]
```diff
@@ -482,10 +499,19 @@ Assertion.prototype = {
...
+          this.includesObject(this.obj, obj)
+            , 'expected ' + this.inspect + ' to include an object equal to ' + i(obj) + (desc ? " | " + desc : "")
+            , 'expected ' + this.inspect + ' to not include an object equal to ' + i(obj) + (desc ? " | " + desc : ""));
+    } else {
+
+      this.assert(
+          ~this.obj.indexOf(obj)
+            , 'expected ' + this.inspect + ' to include ' + i(obj) + (desc ? " | " + desc : "")
+            , 'expected ' + this.inspect + ' to not include ' + i(obj) + (desc ? " | " + desc : ""));
+    }
```
Comment (tj) [inline lib/should.js:512]: whitespace is all weird here, 2 spaces and no need for the newline after else

Comment (rafmagana) [inline lib/should.js:512]: Are you talking about the extra spaces in 506, 507? the commas should be at the same level as `this.assert`?

File: Readme.md (hunk 2 above)
Comment (rafmagana) [inline Readme.md]: oops, you're right, I'll fix that.

File: lib/should.js (hunk 3 above)
Comment (rafmagana) [inline lib/should.js:512]: or at the same level as `this.includesObject` and `~this.obj`?

**Comment by rafmagana** [comment]
what do you think now?

**Comment by tj** [comment]
merged/tweaked as a single commit

**Comment by tj** [comment]
in general though shorter variable names (that still make sense) == better

**Comment by rafmagana** [comment]
cool, thanks!

**Pull request closed without merging**